Sources/SwiftProtobuf/AnyMessageStorage.swift

```diff
@@ -218,7 +218,7 @@ internal class AnyMessageStorage {
     case .contentJSON(let contentJSON, let options):
       // contentJSON requires we have the type available for decoding
       guard let messageType = Google_Protobuf_Any.messageType(forTypeURL: _typeURL) else {
-          throw BinaryEncodingError.anyTypeURLNotRegistered(typeURL: _typeURL)
+        throw SwiftProtobufError.BinaryEncoding.anyTypeURLNotRegistered(typeURL: _typeURL)
```

Comment:
Ok, I think I'd like us to chat about this a bit. @thomasvl and @tbkka your opinions would be valuable.

In Protobuf 1.26, at this call site we used to throw `BinaryEncodingError.anyTranscodeFailure`. I'm inclined to say that where we don't need to change the thrown error type, we probably shouldn't. It's not an API break, so semver doesn't prevent us from making this change, but it does risk causing some trouble with users who actually were trying to catch this error.

However, this would mean these new errors are essentially removed from the tree. How do people feel about that?

Reply:
You might need to go back to the PR, I believe this got added to fix a bug where someone wanted the better visibility into the failure.  So reverting it would likely also mean reopening the bug as we still wouldn't be providing the requested information.

Reply:
Ah, ok, now I follow (I guess my ☕ needed more time 😄).

Historically (in the 1.x releases), we have added error codes, but I don't know that we ever changed a code, and have the existing enums, folks could catch _all_ the errors of that type easily.

But yea, I guess this is different, not only could someone trying to catch this fail due to the change, I guess since it is a new type throw, it would completely escape their `catch` based on the enum. But if we stick to that train of thought pedantically, it means even throwing the new error types is problematic.

I guess I can(am) arguing both outcomes. I'd like to be able to correct errors, add better errors without having to go to a new major version, but I'm not sure where exactly to set the _bar_.

When/How did Nio decide to change the errors over and what was the reasoning behind that? I don't think you all did a major bump when it landed, correct?

Reply:
Generally speaking we don't believe we're beholden to never change the errors we throw. But we need to decide to do it mindfully, rather than doing it without careful consideration. It's _possible_ for us to keep these errors as-is, we just need to decide if we think people are actually planning to catch them.

Reply:
So to summarise, options are:

1. We keep this change, which we've asserted is not-technically-but-de-facto-breaking, meaning users relying on catching the errors originally thrown in 1.26.0 will have to update their code to catch this error instead.
2. We roll this back to the errors thrown in 1.26.0, which means we'll need issues to track https://github.com/apple/swift-protobuf/pull/1527 and any other (if any) changes we've made to make errors more descriptive, and postpone them until(/if) we release a v2.

What are the next steps here? 

Even though I don't know if everyone finds it acceptable to indefinitely postpone having better errors, based on the conversations we've had in the past weeks, it seems to me that we're quite keen on avoiding any sort of breakage - which makes me think we want option 2.
Do we want to wait for @tbkka to weigh in?

Reply:
My lean is probably to do option 2, but I'd like to see if we can get a better advocate for option 1. So far @FranzBusch is the best advocate for it, I wonder if he's up for strengthening his position?

Reply:
I thought it was a good moment to cut over to a new error format all together since we know the enum based errors are not evolvable and we already had a break in `main`. I personally don't share the same strong feeling about changing the errors that we throw since we never promised that we do this.
However, I am also not feeling strong enough to fight you all on this 😀 My main goal right now is to get `main` back into a releasable state without requiring a major. Let's get to that point as fast as possible and then we can re-litigate the error discussion.

Reply:
Should we open a new issue to reference all the things being reverted to provide a clean clearing house of all the errors we'll want to ensure we change whenever we finally do this change?  i.e. - I don't want to accidentally lose a minor detail that we did work out and fail to recreate it whenever we do make the changes.

Reply:
Seems sensible to me.

Reply:
Okay, just made the change we discussed. I believe we're fully non-breaking now.

I'll open an issue listing the things we've reverted from this and other PRs.

Reply:
https://github.com/apple/swift-protobuf/issues/1657